include/swift/Option/FrontendOptions.td

```diff
@@ -469,6 +469,10 @@ def disable_deserialization_recovery :
   Flag<["-"], "disable-deserialization-recovery">,
   HelpText<"Don't attempt to recover from missing xrefs (etc) in swiftmodules">;
 
+def enable_experimental_regex :
+  Flag<["-"], "enable-experimental-regex">,
+  HelpText<"Enable experimental string processing">;
+
```

Comment:
@rxwei mentioned that when we have a `_Regex` module available, it would make sense for that to also set the feature flag (similar to differentiable). At that point I don't know if we'd promote this from a frontend option to something that spans the runtime.